geth-utils/gethutil/mpt/trie/stacktrie.go

```diff
@@ -631,33 +662,40 @@ func (st *StackTrie) UpdateAndGetProofs(db ethdb.KeyValueReader, list types.Deri
 	for i := 1; i < list.Len() && i <= 0x7f; i++ {
```

Comment:
No change is requested here. I just comment to inform the future reviewers.

I saw the duplicated code chunks in L662's loop body, L674's condition body, and L684's loop body.
I was wondering why the duplication but the I saw your comment on L651, which says Geth is also implemented in the same style.